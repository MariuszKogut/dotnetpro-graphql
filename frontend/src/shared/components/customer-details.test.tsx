import React from 'react'
import { render } from '@testing-library/react'
import CustomerDetails from './customer-details'

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}))

describe('CustomerDetails', () => {
  it('renders loading state correctly', () => {
    const { container } = render(<CustomerDetails />)
    expect(container).toMatchSnapshot()
  })

  it('renders insert form correctly', () => {
    // ToDo: insert
    const { container } = render(<CustomerDetails />)
    expect(container).toMatchSnapshot()
  })

  it('renders edit form correctly', () => {
    // ToDo: edit
    const { container } = render(<CustomerDetails />)
    expect(container).toMatchSnapshot()
  })
})
