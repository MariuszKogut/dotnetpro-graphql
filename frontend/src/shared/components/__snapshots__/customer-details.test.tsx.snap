// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDetails renders edit form correctly 1`] = `
<div>
  <div
    class="fade alert alert-info show"
    role="alert"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Kunde wird geladen...
      </span>
    </div>
  </div>
</div>
`;

exports[`CustomerDetails renders insert form correctly 1`] = `
<div>
  <div
    class="fade alert alert-info show"
    role="alert"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Kunde wird geladen...
      </span>
    </div>
  </div>
</div>
`;

exports[`CustomerDetails renders loading state correctly 1`] = `
<div>
  <div
    class="fade alert alert-info show"
    role="alert"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Kunde wird geladen...
      </span>
    </div>
  </div>
</div>
`;
